builtin/logical/pki/cert_util.go

```diff
@@ -226,10 +226,18 @@ func fetchCertBySerial(ctx context.Context, b *backend, req *logical.Request, pr
 
 	// Update old-style paths to new-style paths
 	certEntry.Key = path
+	certsCounted := b.certsCounted.Load()
 	if err = req.Storage.Put(ctx, certEntry); err != nil {
 		return nil, errutil.InternalError{Err: fmt.Sprintf("error saving certificate with serial %s to new location", serial)}
 	}
 	if err = req.Storage.Delete(ctx, legacyPath); err != nil {
+		// If we fail here, we have an extra (copy) of a cert in storage, add to metrics:
+		switch {
+		case strings.HasPrefix(prefix, "revoked/"):
+			b.incrementTotalRevokedCertificatesCount(certsCounted, path)
```

Comment:
Sorry I missed this in the original review, but I believe we need to be passing in just the serial numbers, not the entire paths, within the `incrementTotalRevokedCertificatesCount` and `incrementTotalCertificatesCount` calls. If I understand correctly we will never match as the list operation for say "/certs" only returns the key name not the entire path and key above in the `initializeStoredCertificateCounts` function.

Reply:
Or stripping out the `revoked` and/or `certs` path prefix within `incrementTotalRevokedCertificatesCount` and `incrementTotalCertificatesCount`

Reply:
We will match (the test passes), because of the slice indexing (line 611 of backend.go) right now.  I can see why stripping out the `revoked` and or `certs` prefix in each function might be cleaner, though it would also be a bit slower.

Reply:
ah yes oops I missed that slice operation, thanks for pointing it out! 

Perhaps we could make the slicing operation a little more obvious by extracting it out into a var that is reused across the three tests within the for loop, and not across the code base?

Reply:
Okay, I've moved this to the helper function (the other side), slightly less ugly, and it doesn't seem to be performing any slower when I test it.  Does that work?

Reply:
Yup, even better than what I was asking, thank you!